Library/Homebrew/simulate_system.rb

```diff
@@ -19,24 +24,24 @@ class << self
       def with(os: T.unsafe(nil), arch: T.unsafe(nil), &_block)
         raise ArgumentError, "At least one of `os` or `arch` must be specified." if !os && !arch
 
-        old_os = self.os
-        old_arch = self.arch
+        if self.os || self.arch
+          raise "Cannot simulate#{os&.inspect&.prepend(" ")}#{arch&.inspect&.prepend(" ")} while already " \
+                "simulating#{self.os&.inspect&.prepend(" ")}#{self.arch&.inspect&.prepend(" ")}."
+        end
```

Comment:
Because it shouldn't happen and makes no sense to simulate a system while already simulating another system. At least I cannot think of any use case where you would need to do this.

Reply:
I agree that it'd be nicer to avoid nesting if we can.

Reply:
I am pretty sure we can get rid of `HOMEBREW_SIMULATE_MACOS_ON_LINUX` eventually. Nested simulation might in fact be an indication that it's not needed anymore for those commands that are affected.

Reply:
I'd prefer to get rid of it entirely before we start doing this. This change just makes the gap between `os` and `current_os` larger than it already is, and that's very confusing, especially given the way that they're named.

Reply:
> gap between `os` and `current_os` larger than it already is

Not quite sure what gap you are seeing here. To me it's less confusing if `os` only returns what is actually stored in `@os`.

Reply:
> This change means that `#os` can be `nil` but you're actually simulating macOS.

We aren't using `Homebrew::SimulateSystem.os` anywhere except inside `SimulateSystem` though, so that distinction doesn't matter.

Reply:
I disagree that making the API more confusing doesn't matter.